Firestore/core/src/firebase/firestore/model/field_value.h

```diff
@@ -183,14 +151,12 @@ class FieldValue {
   static FieldValue FromGeoPoint(const GeoPoint& value);
   static FieldValue FromArray(const std::vector<FieldValue>& value);
   static FieldValue FromArray(std::vector<FieldValue>&& value);
-  static FieldValue FromMap(
```

Comment:
Could we preserve these but have them return `ObjectValue` instances instead?

Or maybe preserve these as a way to create a FieldValue from a map in cases where you don't explicitly need the ObjectValue wrapper? Doing so would likely reduce a lot of the churn in the tests.

Reply:
FWIW, the java implementation moves all of this to the ObjectValue class... but since an OV is-a FV, you kinda get the behaviour you're suggesting for free. I've implemented your second suggestion. (This is moving even closer to an inheritance design...)